lib/extension/homeassistant.ts

```diff
@@ -1678,4 +1749,158 @@ export default class HomeAssistant extends Extension {
+            // Sensors:
+            {
+                type: 'sensor',
+                object_id: 'version',
+                mockProperties: [],
+                discovery_payload: {
+                    name: 'Version',
+                    icon: 'mdi:zigbee',
+                    entity_category: 'diagnostic',
+                    state_topic: true,
+                    state_topic_postfix: 'info',
+                    value_template: '{{ value_json.version }}',
+                },
+
+            },
+            {
+                type: 'sensor',
+                object_id: 'coordinator_version',
+                mockProperties: [],
+                discovery_payload: {
+                    name: 'Coordinator version',
+                    icon: 'mdi:chip',
+                    entity_category: 'diagnostic',
+                    enabled_by_default: false,
+                    state_topic: true,
+                    state_topic_postfix: 'info',
+                    value_template: '{{ value_json.coordinator.meta.revision }}',
+                },
+
+            },
+            {
+                type: 'sensor',
+                object_id: 'network_map',
+                mockProperties: [],
+                discovery_payload: {
+                    name: 'Network map',
+                    entity_category: 'diagnostic',
+                    enabled_by_default: false,
+                    state_topic: true,
+                    state_topic_postfix: 'response/networkmap',
+                    value_template: '{{ now().strftime(\'%Y-%m-%d %H:%M:%S\') }}',
+                    json_attributes_topic: `${baseTopic}/response/networkmap`,
+                    json_attributes_template: '{{ value_json.data.value | tojson }}',
+                },
+            },
+            {
+                type: 'sensor',
+                object_id: 'permit_join_timeout',
+                mockProperties: [],
+                discovery_payload: {
+                    name: 'Permit join timeout',
+                    device_class: 'duration',
+                    unit_of_measurement: 's',
+                    entity_category: 'diagnostic',
+                    state_topic: true,
+                    state_topic_postfix: 'info',
+                    value_template: '{{ int(value_json.permit_join_timeout, default="unknown") }}',
+                    expire_after: 2,
+                },
+            },
+
+            // Switches.
+            {
+                type: 'switch',
+                object_id: 'permit_join',
+                mockProperties: [],
+                discovery_payload: {
+                    name: 'Permit join',
+                    icon: 'mdi:human-greeting-proximity',
+                    state_topic: true,
+                    state_topic_postfix: 'info',
+                    value_template: '{{ value_json.permit_join | lower }}',
+                    command_topic: `${baseTopic}/request/permit_join`,
+                    payload_on: 'true',
```

Comment:
Hello, great work! Maybe it's worth adding a 255s timeout there, as is done in the Z2M frontend? I think it's more secure. @Koenkk 

```suggestion
                    command_topic: `${baseTopic}/request/permit_join`,
                    payload_on: '{"value": true, "time": 255}',
```

Reply:
Since this is a home automation system, it's as "secure" as you make your automations. There is a more involved discussion in https://github.com/Koenkk/zigbee2mqtt/discussions/20468#discussioncomment-8005781 (with wildly differing opinions, I don't see a clear consensus either way).